src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThreadGroup.java

```diff
@@ -332,7 +332,19 @@ private boolean run() throws IOException {
                         result = ((FlowInterruptedException) error).getResult();
                     }
                     execution.setResult(result);
-                    t.head.get().addAction(new ErrorAction(error));
+
+                    try {
+                        t.head.get().addAction(new ErrorAction(error));
+                    } catch (Exception e) {
+                        // considered caused for failed to serialize the error.
```

Comment:
Better to consolidate duplicated code into one `ErrorAction.addTo(FlowNode)` utility method.